From 5b58d2dc04f442d4bc00e9aee544b464d043c4a2 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Tue, 27 Sep 2016 15:03:13 +0200
Subject: [PATCH] Revert "nm: Move the D-Bus policy to charon-nm"

This reverts commit 916cd5d7ca35f37e2db81da7db07e12fa96ab373.

We need this until charon-nm >= 5.5.1 is released.
---
 Makefile.am                |  7 +++++--
 nm-strongswan-service.conf | 15 +++++++++++++++
 2 files changed, 20 insertions(+), 2 deletions(-)
 create mode 100644 nm-strongswan-service.conf

diff --git a/Makefile.am b/Makefile.am
index 43f31e7..749a0bd 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -3,6 +3,9 @@ AUTOMAKE_OPTIONS = foreign
 SUBDIRS = properties auth-dialog po
 
 nmvpnservicedir = $(prefix)/lib/NetworkManager/VPN
+
+dbusservicedir = $(sysconfdir)/dbus-1/system.d
+dbusservice_DATA = nm-strongswan-service.conf
 nmvpnservice_DATA = nm-strongswan-service.name
 
 @INTLTOOL_DESKTOP_RULE@
@@ -34,8 +37,8 @@ nm-strongswan-service.name: $(srcdir)/nm-strongswan-service.name.in
 	    -e 's|[@]NM_PLUGINDIR_ABS[@]/|$(nm_plugindir_abs)|g' \
 	    -e 's|[@]CHARON[@]|$(charon)|' $< >$@
 
-EXTRA_DIST = \
-    nm-strongswan-service.name.in  \
+EXTRA_DIST = nm-strongswan-service.name.in \
+    $(dbusservice_DATA)  \
     $(appdata_in_files)  \
     $(appdata_DATA)      \
     intltool-extract.in  \
diff --git a/nm-strongswan-service.conf b/nm-strongswan-service.conf
new file mode 100644
index 0000000..a630f34
--- /dev/null
+++ b/nm-strongswan-service.conf
@@ -0,0 +1,15 @@
+<!DOCTYPE busconfig PUBLIC
+ "-//freedesktop//DTD D-BUS Bus Configuration 1.0//EN"
+ "http://www.freedesktop.org/standards/dbus/1.0/busconfig.dtd">
+<busconfig>
+	<policy user="root">
+		<allow own="org.freedesktop.NetworkManager.strongswan"/>
+		<allow send_destination="org.freedesktop.NetworkManager.strongswan"/>
+		<allow send_interface="org.freedesktop.NetworkManager.strongswan"/>
+	</policy>
+	<policy context="default">
+		<deny own="org.freedesktop.NetworkManager.strongswan"/>
+		<deny send_destination="org.freedesktop.NetworkManager.strongswan"/>
+	</policy>
+</busconfig>
+
-- 
2.7.4

